DQM/L1TMonitor/src/L1TStage2uGT.cc

```diff
@@ -282,7 +391,75 @@ void L1TStage2uGT::analyze(const edm::Event& evt, const edm::EventSetup& evtSetu
            } // end of uGtAlgs
         } // end of BX
      } // selecting LastCollisionInTrain
-  } // end of uGTAlgs = 1
+
+
+   if(algoBitFirstBxInTrain_ != -1 && itr->getAlgoDecisionInitial(algoBitFirstBxInTrain_)) {
+     for(int ibx = uGtAlgs->getFirstBX(); ibx <= uGtAlgs->getLastBX(); ++ibx) {
+       for(auto itr2 = uGtAlgs->begin(ibx); itr2 != uGtAlgs->end(ibx); ++itr2) {
+         for(unsigned int algo=0; algo<prescaledAlgoBitName_.size(); algo++) { 
+           if(itr2->getAlgoDecisionInitial(prescaledAlgoBitName_.at(algo).second)) { 
+             prescaled_first_collision_run_->Fill(ibx, prescaledAlgoBitName_.at(algo).second);
+             for(int ibx2 = uGtAlgs->getFirstBX(); ibx2 <= uGtAlgs->getLastBX(); ++ibx2) {
+               den_prescaled_first_collision_run_->Fill(ibx2,prescaledAlgoBitName_.at(algo).second);
+             }
+           }
+         }
+         for(unsigned int algo=0; algo<unprescaledAlgoBitName_.size(); algo++) {
+           if(itr2->getAlgoDecisionInitial(unprescaledAlgoBitName_.at(algo).second)) {
+             unprescaled_first_collision_run_->Fill(ibx,unprescaledAlgoBitName_.at(algo).second);
+             for(int ibx2 = uGtAlgs->getFirstBX(); ibx2 <= uGtAlgs->getLastBX(); ++ibx2) {
+               den_unprescaled_first_collision_run_->Fill(ibx2,unprescaledAlgoBitName_.at(algo).second);
+             }
+           }
+         }
+       }
+     }
+   }
+   if(algoBitLastBxInTrain_ != -1 && itr->getAlgoDecisionInitial(algoBitLastBxInTrain_)) {
+     for(int ibx = uGtAlgs->getFirstBX(); ibx <= uGtAlgs->getLastBX(); ++ibx) {
+       for(auto itr2 = uGtAlgs->begin(ibx); itr2 != uGtAlgs->end(ibx); ++itr2) {
+         for(unsigned int i=0; i<prescaledAlgoBitName_.size(); i++) {
+           if(itr2->getAlgoDecisionInitial(prescaledAlgoBitName_.at(i).second)) {
+             prescaled_last_collision_run_->Fill(ibx, prescaledAlgoBitName_.at(i).second);
+             for(int ibx2 = uGtAlgs->getFirstBX(); ibx2 <= uGtAlgs->getLastBX(); ++ibx2) {
+               den_prescaled_last_collision_run_->Fill(ibx2,prescaledAlgoBitName_.at(i).second);
+             }
+           }
+         } 
+         for(unsigned int i=0; i<unprescaledAlgoBitName_.size(); i++) {
+           if(itr2->getAlgoDecisionInitial(unprescaledAlgoBitName_.at(i).second)) {
+             unprescaled_last_collision_run_->Fill(ibx,unprescaledAlgoBitName_.at(i).second);
+             for(int ibx2 = uGtAlgs->getFirstBX(); ibx2 <= uGtAlgs->getLastBX(); ++ibx2) {
+               den_unprescaled_last_collision_run_->Fill(ibx2,unprescaledAlgoBitName_.at(i).second);
```

Comment:
The filling of the y axis value is wrong. You have to fill the bin number not the algo bit.
`den_unprescaled_last_collision_run_->Fill(ibx2, i+1);`